src/concurrency/Sync.java

```diff
@@ -24,10 +23,18 @@ public void run() {
     public static void main(String[] args)  {
         Sync syncThread = new Sync("sync thread");
         syncThread.start();
+        try {
+            Thread.sleep(1000);
```

Comment:
This approach is VERY costly and risky.  There is an assumption that it will never take more than 1 second for the list to be generated.  If the system happens to be sluggish for some reason, you now return prematurely.  Additionally, if the system completes the task in .25 seconds, it is still going to wait (for nothing) for another .75 seconds before returning.  There are better Thread related features available which will wait until the Thread has finished its work, regardless of how long it takes.